exercises/difference-of-squares/example.pony

```diff
@@ -12,4 +12,4 @@ primitive Squares
       .fold[USize](0, {(acc: USize, x: USize): USize => acc + x })
 
   fun difference(n: USize): USize =>
-    square_of_sums(n) - sum_of_squares(n)
+    square_of_sum(n) - sum_of_squares(n)
```

Comment:
The sum of the squares and the square of the sums is different than saying the sum of the square and the square of the sum.  So I do not believe this change is correct.